---
title: "Wastewater AMR in Latvia"
author: "Edgars Liepa"
date: "26/03/2025"
output:
  pdf_document: default
  toc: true
  toc_depth: 2
---

```{r, lib, warning=FALSE}
library("phyloseq")
library("vegan")
library("dplyr")
suppressPackageStartupMessages(library(microViz))
library("ggplot2")
library(jsonlite)
library(patchwork)
library(stringr)
library(tidyr)
```

For resistance gene annotations CARD database was used.

```{r, paths}
CARDDB = "~/DB/CARD_02_2024/card.json"

METADATA_PATH =  "../sampleMetadata.csv"
metadata <- read.csv(METADATA_PATH, header = TRUE, colClasses = c(Date = "character", Dairy_farming="character", Meat_production="character", Metal_processing="character", Washrooms="character"))

```


## ARG abundance

Read data resistance data.


AMR genes were classified in metagenome assemblies using Resistance gene finder.

Then genes were quantified by aligning short read sequencing reads to the assemblies
and sequenceing read overlap was counted in positions with ht-seq. 


```{r, read_amr}
ARG_SCAFFOLDS_F = "../Resistance_genes/AMR_genes_RGI_scaffolds_filtered.csv"
arg_t <- read.csv(ARG_SCAFFOLDS_F, header = TRUE, row.names = 1)


tax_t <- data.frame(Tax = rownames(arg_t), ARG = rownames(arg_t))
  # tax_t <- sanitizeRowNames(tax_t)
rownames(tax_t) <- tax_t$Tax

ARG_TAX <- tax_table(as.matrix(tax_t[, c('Tax', 'ARG')]))

ps_amr <- phyloseq(otu_table(arg_t, taxa_are_rows = TRUE), sample_data(metadata), ARG_TAX)

# Fix Sample names
# Create a named vector for mapping
name_mapping <- setNames(metadata$Sample, rownames(metadata))

# Use the mapping to rename the columns
sample_names(ps_amr) <- name_mapping[sample_names(ps_amr)]

```

Display amount of sequencing reads alligning to a AMR genes found in contigs.

```{r, plot_amr}

plot_bar(ps_amr)

options(repr.plot.width=4, repr.plot.height=4)
hist(log10(sample_sums(ps_amr)), breaks=50, main="Sample size distribution", xlab="Sample size (log10)", ylab="Frequency", col="#007c80")

```
### Filtering 

Since we removed some samples from bacterial anlysis, we have to remove them also here.    

```{r, hist_amr}
ps_amr_good = subset_samples(ps_amr, sample_data(ps_amr)$Sample != "Salaspils.2")
ps_amr_good = subset_samples(ps_amr_good, sample_data(ps_amr_good)$Sample != "Talsi.3")

hist(log10(sample_sums(ps_amr_good)), breaks=50, main="Sample size distribution", xlab="Sample size (log10)", ylab="Frequency", col="#007c80")
```
Now we remove singleton reads that are assigned to genes with only one alligned read across all samples.

```{r, filter_amr}
# Less strict filterring option
# remove singletons. 
ps_scaffolds_filtered = phyloseq::filter_taxa(ps_amr_good, function(x) sum(x) > 1, prune=TRUE)


max_difference = max(sample_sums(ps_scaffolds_filtered))/min(sample_sums(ps_scaffolds_filtered))

# The max difference in sequencing depth between largest and smallest sample
max_difference

```

```{r, amr_depth}
boxplot(sample_sums(ps_scaffolds_filtered), main="Sequencing depth across samples removed singletons", xlab="", ylab="Number of reads", col="#a6093d")
text(y=boxplot.stats(sample_sums(ps_scaffolds_filtered))$stats, labels=boxplot.stats(sample_sums(ps_scaffolds_filtered))$stats, x=1.25)
```

Print the dimensions of the filtered data
Number of genes found in samples after filtering

```{r, amr_size}
dim(data.frame(otu_table(ps_scaffolds_filtered)))
```


Print the total sequence count after filtering
Number of sequences in all samples

```{r, amr_size2}
sum(data.frame(otu_table(ps_scaffolds_filtered)))
```


Calculate and print the percentage of sequences dropped from the original dataset
```{r, calc_drop}
original_seq_count = sum(data.frame(otu_table(ps_amr_good)))
filtered_seq_count = sum(data.frame(otu_table(ps_scaffolds_filtered)))
seq_dropped_percentage = ((original_seq_count - filtered_seq_count) / original_seq_count) * 100
```


```{r, seq_drop}
# Sequence % dropped from the dataset:
seq_dropped_percentage
```


Show AMR gene distribution across samples

```{r, abundance_plot, warning=FALSE}
ps_scaffolds_filtered %>%
    ps_mutate(Group = factor(sample_data(ps_scaffolds_filtered)$City)) %>%
    tax_agg("ARG") %>%
    ps_seriate(dist = "bray", method = "OLO_ward") %>%
    comp_barplot(tax_level = "ARG", 
                 sample_order = rownames(sample_data(ps_scaffolds_filtered)[order(sample_data(ps_scaffolds_filtered)$Sample), ]), 
                 n_taxa = 10, 
                 label = "Sample") +
    theme(axis.text.x = element_text(angle = 90, hjust = 1), legend.position = "bottom")+
    facet_grid(~Group, scales = "free", space = "free")
```

### CARD DB gene anotations

Using CARD DB we take reconstructed genes to assign for witch AMR group, AMR familie and AMR category they belong.

```{r, load_card, results='hide'}

source("~/ARG_waste_water/src/comperative_metagen_definitions.r")

card_tax_t <- NA
card_tax_t <- process_card_data(arg_t)

#
# SEPERATE DRUG CLASSES INTO INDIVIDUAL NAMES
#

# Assuming tax_t is your data frame and DrugClass is the column of interest
# unique_drug_classes
unique_drug_classes <- card_tax_t %>%
  # Separate the DrugClass column into rows by splitting on ";"
  separate_rows(DrugClass, sep = ";") %>%
  # Select unique DrugClass names
  distinct(DrugClass) %>%
  # Pull the DrugClass column to get a vector of unique drug class names
  pull(DrugClass)




#unique_amr_families
unique_amr_families <- card_tax_t %>%
  # Separate the DrugClass column into rows by splitting on ";"
  separate_rows(AMRGeneFamily, sep = ";") %>%
  # Select unique DrugClass names
  distinct(AMRGeneFamily) %>%
  # Pull the DrugClass column to get a vector of unique drug class names
  pull(AMRGeneFamily)

#unique_amr_categories
unique_amr_categories <- card_tax_t %>%
  # Separate the DrugClass column into rows by splitting on ";"
  separate_rows(AntibioticCategory, sep = ";") %>%
  # Select unique DrugClass names
  distinct(AntibioticCategory) %>%
  # Pull the DrugClass column to get a vector of unique drug class names
  pull(AntibioticCategory)

```

Create Objects for AMR categories.

```{r, create_card_objects}
drug_class_counts <- create_drug_class_counts(arg_t, card_tax_t, unique_drug_classes)

# Create the AMR families counts
amr_families_counts <- create_amr_families_counts(arg_t, card_tax_t, unique_amr_families)

# Create the AMR categories counts
amr_categories_counts <- create_amr_categories_counts(arg_t, card_tax_t, unique_amr_categories)

# Create the taxonomy table
tax_t_drugs <- create_taxonomy_table(card_tax_t)

physeq_drug_class <- createPhyloseqObject(drug_class_counts, METADATA_PATH)
physeq_amr_families <- createPhyloseqObject(amr_families_counts, METADATA_PATH)
physeq_amr_categories <- createPhyloseqObject(amr_categories_counts, METADATA_PATH)
```

Show to what Drug classes resistance genes found in metagenome are resistant to.

```{r, warning=FALSE}
physeq_amr_categories %>%
    ps_mutate(Group = factor(sample_data(physeq_amr_categories)$City)) %>%
    tax_agg("ARG") %>%
    ps_seriate(dist = "bray", method = "OLO_ward") %>%
    comp_barplot(tax_level = "ARG", 
                 sample_order = rownames(metadata[order(metadata$Sample), ]), 
                 n_taxa = 10, 
                 label = "Sample", interactive = TRUE) +
    theme(axis.text.x = element_blank(), axis.ticks.x = element_blank(), legend.position = "bottom")+
    facet_grid(~Group, scales = "free", space = "free")+
    labs(y = "Relative Abundance", fill = "AMR categories")
```


```{r, drug_class_pl}
physeq_drug_class %>%
    ps_mutate(Group = factor(sample_data(physeq_drug_class)$City)) %>%
    tax_agg("ARG") %>%
    ps_seriate(dist = "bray", method = "OLO_ward") %>%
    comp_barplot(tax_level = "ARG", 
                 sample_order = rownames(metadata[order(metadata$Sample), ]), 
                 n_taxa = 10, 
                 label = "Sample", interactive = TRUE) +
    theme(axis.text.x = element_text(angle = 90, hjust = 1), legend.position = "bottom")+
    facet_grid(~Group, scales = "free", space = "free")

```

```{r, family_pl, fig.width=10, fig.height=4.5}
physeq_amr_families %>%
    ps_mutate(Group = factor(sample_data(physeq_amr_families)$City)) %>%
    tax_agg("ARG") %>%
    ps_seriate(dist = "bray", method = "OLO_ward") %>%
    comp_barplot(tax_level = "ARG", 
                 sample_order = rownames(metadata[order(metadata$Sample), ]), 
                 n_taxa = 10, 
                 label = "Sample", interactive = TRUE) +
    theme(
      axis.text.x = element_blank(), 
      axis.ticks.x = element_blank(), 
      legend.position = "bottom"
      )+
    guides(fill = guide_legend(ncol = 3))+
    facet_grid(~Group, scales = "free", space = "free")

```

```{r, combined_pl, fig.height=12, fig.width=14}

# Plot 1 (physeq_drug_class - renamed for clarity)
p1 <- physeq_drug_class %>%
  ps_mutate(Group = factor(sample_data(physeq_drug_class)$City)) %>%
  tax_agg("ARG") %>%
  ps_seriate(dist = "bray", method = "OLO_ward") %>%
  comp_barplot(tax_level = "ARG",
               sample_order = rownames(metadata[order(metadata$Sample), ]),
               n_taxa = 10,
               label = "Sample", interactive = FALSE) + # Turn off interactive for combination
  theme(
        legend.position = "right",
        axis.title.x = element_blank(),
        axis.text.x = element_blank(),
        axis.ticks.x = element_blank(),
        axis.text.y = element_text(size = 14),
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 14),
        strip.text = element_text(size = 14),
  ) +
  facet_grid(~Group, scales = "free", space = "free") +   
  labs(fill = "Abtibiotic Class")



# Plot 2 (physeq_amr_families - renamed for clarity)
p2 <- physeq_amr_families %>%
  ps_mutate(Group = factor(sample_data(physeq_amr_families)$City)) %>%
  tax_agg("ARG") %>%
  ps_seriate(dist = "bray", method = "OLO_ward") %>%
  comp_barplot(tax_level = "ARG",
               sample_order = rownames(metadata[order(metadata$Sample), ]),
               n_taxa = 10,
               label = "Sample", interactive = FALSE) +  # Turn off interactive
  theme(
        axis.text.x = element_text(angle = 90, hjust = 1, size = 12), # Smaller x-axis text
        axis.ticks.x = element_blank(),
        axis.text.y = element_text(size = 12),
        strip.text = element_blank(),
        legend.position = "right",
        axis.title.x = element_blank(),
        legend.text = element_text(size = 14),
        legend.title = element_text(size = 12)
        
  ) + # Remove x-axis title
  facet_grid(~Group, scales = "free", space = "free") +
  ylab("Relative Abundance") +   # Clear y-axis label
  labs(fill = "AMR family")



# Combine the plots vertically using patchwork
combined_plot <- p1 / p2

print(combined_plot)
```

```{r, save_divers_pl, echo=FALSE}

# Save as PNG
ggsave("../plots/arg_diversity.png", plot = combined_plot, width = 24, height = 12, units = "in", dpi = 300)

# Save as SVG
ggsave("../plots/arg_diversity.svg", plot = combined_plot, width = 24, height = 12, units = "in")

```



Show gene distribution



```{r, ordination_plot}

GP1 = transform_sample_counts(ps_scaffolds_filtered, function(x) 1E6 * x/sum(x))

arg.sum = tapply(taxa_sums(GP1), tax_table(GP1)[, "ARG"], sum, na.rm=TRUE)
top10args = names(sort(arg.sum, TRUE))[1:10]
GP1 = prune_taxa((tax_table(GP1)[, "ARG"] %in% top10args), GP1)

GP.ord <- ordinate(GP1, "NMDS", "bray")
GP.ord2 <- ordinate(ps_scaffolds_filtered, "NMDS", "bray")

plot_ordination(GP1, GP.ord2, type="taxa", color = "ARG", title="ARGs") + theme(legend.position = "bottom")

```


### Normalization


Normalization by subsampling (rarefaction)

Display rarefication curves.

```{r, rare_curve_amr}
tab <- otu_table(ps_scaffolds_filtered)
class(tab) <- "matrix"
tab <- t(tab)

rarecurve(tab, step=10, lwd=2, ylab="OTU",  label=F)

```

Rarefication if needed.

```{r, rare_amr}
# find minimal deapth to rarefy to
df=as.data.frame(sample_sums(ps_scaffolds_filtered))
head(df[order( df[,1] ),],1)

ps_amr_rare = rarefy_even_depth(ps_scaffolds_filtered, sample.size=820, replace=FALSE, rngseed=123, verbose=FALSE)

```


Normalization by cumulative sum scaling (CSS)

```{r, css_amr}
ps_amr_CSS = microbiomeMarker::normalize(ps_scaffolds_filtered, method="CSS")
```

### Most abundant genes

Show Top 15 most abundant genes and their relative xount

```{r, top_amr}

genes_transformed  = transform_sample_counts(ps_scaffolds_filtered, function(x) x / sum(x) )

abu_table <- as.data.frame(otu_table(genes_transformed))
tax_table <- as.data.frame(tax_table(genes_transformed))
rownames(abu_table) <- tax_table$ARG

total_counts <- rowSums(abu_table)
top_organisms <- sort(total_counts, decreasing = TRUE)
top_organisms <- head(top_organisms, n = 15)

top_organisms

mean_percentage <- apply(abu_table, 1, mean) * 100
std_dev <- sqrt(apply(abu_table, 1, var)) * 100

top_genes <- sort(mean_percentage, decreasing = TRUE) %>% head(n = 15) %>% as.data.frame %>% rownames

# Format output with both mean percentage and standard deviation
formatted_output <- sapply(top_genes, function(gene) {
  sprintf("%s (%.2f%% +/- %.2f%%)", 
          gene, 
          unname(mean_percentage[gene]), 
          unname(std_dev[gene])
          )
})

# Display formatted output
formatted_output
```


Show Drug classes that these genes are affecting

```{r, drug_top}
abu_table <- as.data.frame(otu_table(physeq_drug_class))
tax_table <- as.data.frame(tax_table(physeq_drug_class))
rownames(abu_table) <- tax_table$ARG

total_counts <- rowSums(abu_table)
top_organisms <- sort(total_counts, decreasing = TRUE)
top_organisms <- head(top_organisms, n = 15)

top_organisms

# Calculate the percentage of the top 15 organisms
print("Percentage of top 15 organisms:")
top_organisms/sum(rowSums(abu_table))*100
```


Show most common amr_gene families.

```{r, top_amr2}
abu_table <- as.data.frame(otu_table(physeq_amr_families))
tax_table <- as.data.frame(tax_table(physeq_amr_families))
rownames(abu_table) <- tax_table$ARG

total_counts <- rowSums(abu_table)
top_organisms <- sort(total_counts, decreasing = TRUE)
top_organisms <- head(top_organisms, n = 15)

top_organisms

```

Show most commonly affected drug classes.

```{r}
abu_table <- as.data.frame(otu_table(physeq_amr_categories))
tax_table <- as.data.frame(tax_table(physeq_amr_categories))
rownames(abu_table) <- tax_table$ARG

total_counts <- rowSums(abu_table)
top_organisms <- sort(total_counts, decreasing = TRUE)
top_organisms <- head(top_organisms, n = 15)

top_organisms

```

Calculate mean relative abundance and variance

```{r}

genus_transformed_amr  = transform_sample_counts(physeq_amr_families, function(x) x / sum(x) )

abu_table_arg <- as.data.frame(otu_table(genus_transformed_amr))
tax_table_arg <- as.data.frame(tax_table(genus_transformed_amr))
rownames(abu_table_arg) <- tax_table_arg$Tax

variance <- apply(abu_table_arg, 1, var)
mean_percentage <- apply(abu_table_arg, 1, mean) * 100
std_dev <- sqrt(apply(abu_table_arg, 1, var)) * 100


top_arg <- sort(mean_percentage, decreasing = TRUE) %>% head(n = 15) %>% as.data.frame %>% rownames

# Format output with both mean percentage and standard deviation
formatted_output <- sapply(top_arg, function(arg) {
  sprintf("%s (%.2f%% +/- %.2f%%)", 
          arg, 
          unname(mean_percentage[arg]), 
          unname(std_dev[arg])
          )
})

# Display formatted output
formatted_output

variance <- apply(abu_table_arg["MCR phosphoethanolamine transferase",], 1, var)
mean_percentage <- apply(abu_table_arg["MCR phosphoethanolamine transferase",], 1, mean) * 100
std_dev <- sqrt(apply(abu_table_arg["MCR phosphoethanolamine transferase",], 1, var)) * 100


```



## City

Exploring differences how AMR gene composition is affected by Municipality factor.

### ARG

#### Alpha diversity




```{r, plot_div_amr, fig.width=7.2, fig.height=6}
alph_city <- plot_richness(ps_scaffolds_filtered, x="City", measures=c("InvSimpson", "Shannon"), nrow = 2, sortby = "Shannon") + 
    geom_boxplot() + 
    theme_bw() +
    theme(legend.position = "none", axis.text.x = element_text(angle = 90, hjust = 1))+
  theme(
    legend.position = "right",
    axis.text.x = element_text(angle = 90, hjust = 1, size = 14),
    axis.text.y = element_text(size = 14),
    axis.title = element_text(size = 14),
    legend.text = element_text(size = 19),
    legend.title = element_text(size = 18)
  ) 

alph_city

```



```{r_ save_div_amr, echo=FALSE}
# Save as PNG
ggsave("../plots/arg_alpha_city_2.jpg", plot = alph_city, width = 8, height = 6, units = "in", dpi = 1200)

# Save as SVG
ggsave("../plots/arg_alpha_city_2.svg", plot = alph_city, width = 8, height = 6, units = "in", dpi = 1200)

```

Calculate distance

```{r, fig.width=9, fig.height=5, text.width=10}
# Calculate diversity indexes
alpha_indexes_amr <- estimate_richness(ps_scaffolds_filtered, split = TRUE, c("Shannon", "Simpson", "InvSimpson"))

kruskal.test(alpha_indexes_amr$Shannon ~ sample_data(ps_scaffolds_filtered)$City)
kruskal.test(alpha_indexes_amr$InvSimpson ~ sample_data(ps_scaffolds_filtered)$City)

```

```{r, fig.width=9, fig.height=5, text.width=10}
# Calculate diversity indexes
alpha_indexes_amr <- estimate_richness(ps_scaffolds_filtered, split = TRUE, c("Shannon", "Simpson", "InvSimpson"))

kruskal.test(alpha_indexes_amr$Shannon ~ sample_data(ps_scaffolds_filtered)$City)
kruskal.test(alpha_indexes_amr$InvSimpson ~ sample_data(ps_scaffolds_filtered)$City)

```




Kruskal-Wallis rank sum test show significant differences in overall AMR diversity (Shannon)
but dont show significant differences in how specific AMR dominate between cities (InvSimpson).


Lets show minimal and maximux diversity values.

```{r, min_max}
min(alpha_indexes_amr$InvSimpson)
max(alpha_indexes_amr$InvSimpson)
min(alpha_indexes_amr$Shannon)
max(alpha_indexes_amr$Shannon)
```

Dunes Test 

pinpoint which specific means are significantlt different from the others

```{r, dune_amr_shan}
library(dunn.test)
dunn_results <- dunn.test(alpha_indexes_amr$Shannon, 
          sample_data(ps_scaffolds_filtered)$City, 
          method="bh")
```

```{r, dune_yo}
dunn_results$P.adjusted[dunn_results$P.adjusted < 0.05]
dunn_results$chi2
dunn_results$Z[dunn_results$P.adjusted < 0.05]
```

Shows most significantly different pairs between cities.

```{r, dune_yo2}
dunn_results$comparisons[dunn_results$P.adjusted < 0.05]
```

```{r}
dunn_results <- dunn.test(alpha_indexes_amr$InvSimpson, 
          sample_data(ps_scaffolds_filtered)$City, 
          method="bh")
```


```{r, dune_yo3}
dunn_results$P.adjusted[dunn_results$P.adjusted < 0.05]
dunn_results$chi2
dunn_results$Z[dunn_results$P.adjusted < 0.05]
dunn_results$comparisons[dunn_results$P.adjusted < 0.05]
```

#### Beta diversity

Lets compare how simmilar municipalities are between each other using bray-curtis distance on a NMDS plot.

```{r, beta_amr_city}
GP.ord2 <- ordinate(ps_amr_CSS, "NMDS", "bray")
p1 = plot_ordination(ps_amr_CSS, GP.ord2, type="samples", color = "City", title="City bray-distance NMDS")+
 geom_polygon(aes(fill=City), alpha = 1/2) + geom_point(size=3)

print(p1)
```

Calculate distances

```{r, dist_amr}
tax_bray_amr <- phyloseq::distance(ps_amr_CSS, method = "bray")
```

Lets check variance with ANOVA.

```{r, anova_amr}

anosim(tax_bray_amr, sample_data(ps_amr_CSS)$City, distance = "bray", permutations = 999)

```
PERMANOVA

```{r, permanova_city_amr}

adonis2_rez<-adonis2(tax_bray_amr ~ sample_data(ps_amr_CSS)$City)
print(adonis2_rez)
```

Chcek beta dispersion between City

```{r, disp_amr_city}
beta <- betadisper(tax_bray_amr, sample_data(ps_amr_CSS)$City)
print(anova(beta))
plot(beta)

```
### Population Size


Now look wether population size impact AMR.

```{r, kruskal_popul}

kruskal.test(alpha_indexes_amr$Shannon ~ sample_data(ps_scaffolds_filtered)$Population)
kruskal.test(alpha_indexes_amr$InvSimpson ~ sample_data(ps_scaffolds_filtered)$Population)

```

```{r, div_popul}
plot_richness(ps_scaffolds_filtered, x="Population", title="AMR diversity", measures=c("InvSimpson", "Shannon"), nrow = 2, sortby = "InvSimpson") + 
    geom_boxplot() + 
    theme_bw() +
    theme(legend.position = "none", axis.text.x = element_text(angle = 90, hjust = 1))+
  theme(
    legend.position = "right",
    axis.text.x = element_text(angle = 90, hjust = 1, size = 14),
    axis.text.y = element_text(size = 14),
    axis.title = element_text(size = 14),
    legend.text = element_text(size = 14),
    legend.title = element_text(size = 14)
  )

```

```{r, plot_pop}
beta_popul <- ps_scaffolds_filtered %>%
  dist_calc(dist = "bray", binary = TRUE) %>%
  ord_calc("NMDS") %>%
  ord_plot(
    color = "Population", 
    shape = "City", 
    size = 5,
    alpha = 0.8
  ) +
  scale_shape_manual(values = c(16, 17, 18, 15, 19, 25, 8, 13, 1, 2, 3, 4, 5, 6, 7)) + # Manual shapes for all hospital types
  stat_ellipse(aes(linetype = Population, colour = Population)) +
  theme_bw() +
  scale_color_brewer(palette = "Set1") +
  scale_fill_brewer(palette = "Set1") +
  labs(
    title = "NMDS of Population Profiles",
    x = "NMDS1",
    y = "NMDS2"
  )+
  theme(
    legend.position = "right",
    axis.text.x = element_text(angle = 90, hjust = 1, size = 14),
    axis.text.y = element_text(size = 14),
    axis.title = element_text(size = 14),
    legend.text = element_text(size = 14),
    legend.title = element_text(size = 16)
  ) 


beta_popul
```

```{r, save_plot_nmds_pop, echo=FALSE}
# Save as PNG
ggsave("../plots/arg_beta_pop.png", plot = beta_popul, width = 10, height = 8, units = "in", dpi = 300)

# Save as SVG
ggsave("../plots/arg_beta_pop.svg", plot = beta_popul, width = 10, height = 8, units = "in")

```


```{r, plot_beta_pop}


adonis2_rez<-adonis2(tax_bray_amr ~ sample_data(ps_amr_CSS)$Population)
print(adonis2_rez)

beta <- betadisper(tax_bray_amr, sample_data(ps_amr_CSS)$Population)
print("BETADISPER")
print("Population")
print(anova(beta))
plot(beta)
```

## Hospital

No significance in Regional Hospital.

```{r, wilconx_hosp}
wilcox.test(alpha_indexes_amr$Shannon~sample_data(ps_scaffolds_filtered)$Regional_Hospital,
                     p.adjust.method = "BH")

wilcox.test(alpha_indexes_amr$InvSimpson~sample_data(ps_scaffolds_filtered)$Regional_Hospital,
                     p.adjust.method = "BH")

```

Hospital Types


```{r, kurskal_amr}
kruskal.test(alpha_indexes_amr$Shannon ~ sample_data(ps_scaffolds_filtered)$Hospital_Type)
kruskal.test(alpha_indexes_amr$InvSimpson ~ sample_data(ps_scaffolds_filtered)$Hospital_Type)
```

```{r, richness_h_type, echo=FALSE}
# used for generating plots
r1 <- plot_richness(ps_scaffolds_filtered, x = "Hospital_Type",
              measures=c("Shannon", "InvSimpson"), nrow = 2, sortby = "Shannon") +
              geom_boxplot() + geom_point(size=1, alpha=0.5)  + labs(
    title = "AMR Hospital type diversity")  +
    theme(
        axis.text.x = element_text(angle = 90, hjust = 1, size = 18),
        axis.ticks.x = element_blank(),
        strip.text = element_text(size = 18),
        axis.text.y = element_text(size = 18),
        title = element_text(size = 22)
  )

r2 <- plot_richness(ps_bact_filtered_species, x = "Hospital_Type",
              measures=c("Shannon", "InvSimpson"), nrow = 2, sortby = "Shannon") +
              geom_boxplot() + geom_point(size=1, alpha=0.5)   + labs(
    title = "Species Hospital type diversity")+
  theme(
        axis.text.x = element_text(angle = 90, hjust = 1, size = 18), # Smaller x-axis text
        axis.ticks.x = element_blank(),
        axis.text.y = element_text(size = 18),
        strip.text = element_text(size = 18),
        legend.position = "right",
        title = element_text(size = 22)
  )


r_plot <- r1 + r2

r_plot
```

```{r, save_hosp_rich, echo=FALSE}

# # Save as PNG
ggsave("../plots/arg_alpha_hospital.jpg", plot = r_plot, width = 24, height = 12, units = "in", dpi = 1200)
# 
# # Save as SVG
# ggsave("../plots/arg_alpha_hospital.svg", plot = r_plot, width = 24, height = 12, units = "in")

```

```{r, beta_h_type}
GP.ord2 <- ordinate(ps_amr_CSS, "NMDS", "bray")
p1 = plot_ordination(ps_amr_CSS, GP.ord2, type="samples", color = "Hospital_Type", title="Hospital_Type bray-distance NMDS")+
 stat_ellipse(aes(linetype = Regional_Hospital, colour = Regional_Hospital)) + geom_point(size=3)

print(p1)
```


```{r, beta_h_type2}
adonis2_rez<-adonis2(tax_bray_amr ~ sample_data(ps_amr_CSS)$Hospital_Type)
print(adonis2_rez)
```


Check BETADISPER between Hospital_Types

```{r, beta_h_type3}
beta <- betadisper(tax_bray_amr, sample_data(ps_amr_CSS)$Hospital_Type)
print(anova(beta))
plot(beta)
```


```{r, dune_h_typ}
dunn_results <- dunn.test(alpha_indexes_amr$Shannon, 
          sample_data(ps_scaffolds_filtered)$Hospital_Type, 
          method="bh")
```


```{r, dune_h_typ2}
dunn_results$chi2
dunn_results$P.adjusted[dunn_results$P.adjusted < 0.05]
dunn_results$Z[dunn_results$P.adjusted < 0.05]
dunn_results$comparisons[dunn_results$P.adjusted < 0.05]
```


```{r, dune_h}
dunn_results <- dunn.test(alpha_indexes_amr$InvSimpson, 
          sample_data(ps_scaffolds_filtered)$Hospital_Type, 
          method="bh")
```


```{r, dune_h2}
dunn_results$P.adjusted[dunn_results$P.adjusted < 0.05]
dunn_results$chi2
dunn_results$Z[dunn_results$P.adjusted < 0.05]
dunn_results$comparisons[dunn_results$P.adjusted < 0.05]
```


#### SIAMCAT

Using siamcat we look for genes that are explaining most of the variations by factor groups.

```{r, tax_glom}
library(SIAMCAT)
relab_ps = transform_sample_counts(ps_scaffolds_filtered, function(x) x/sum(x))

psglom = tax_glom(relab_ps, "ARG")
```

```{r, siam_plots}
create_siamcat_plot <- function(psg, label_column, case_value, output_prefix) {
  # Create label
  sc_label <- create.label(meta=sample_data(psg), label=label_column, case=case_value)
  
  # Create SIAMCAT object
  siamcat_o <- siamcat(phyloseq=psg, label=sc_label)
  
  
  # Filter features
  siamcat_o <- filter.features(siamcat_o, filter.method='abundance', cutoff=1e-03)
  siamcat_o <- filter.features(siamcat_o, filter.method='prevalence', cutoff=0.05, feature.type='filtered')
  
  # Check associations
  siamcat_o <- check.associations(siamcat_o)
  
  association.plot(siamcat_o, panels=c("fc", "prevalence"), prompt = FALSE, verbose = 0)
  # Final association plot
  svg(paste0("../plots/",output_prefix, "_final_association.svg"), width=12, height=7)
  association.plot(siamcat_o, panels=c("fc", "prevalence"), prompt = FALSE, verbose = 0)
  dev.off()
  cat("plot saved to", paste0(output_prefix, "_final_association.svg"), "\n")
  
  return(siamcat_o)
}

```

Here most differentially found genes are plotted

```{r, assoc_hosp}
label_case_pairs <- list(
  list(label = "Hospital_Type", case = "0"),
  list(label = "Hospital_Type", case = "2"),
  list(label = "Hospital_Type", case = "3"),
  list(label = "Hospital_Type", case = "4"),
  list(label = "Hospital_Type", case = "Specilised"),
  list(label = "Hospital_Type", case = "Other"),
  list(label = "Hospital_Type", case = "Red_cross"),
  list(label = "Hospital_Type", case = "Branch")
  # Add more pairs as needed
)

#psglom = tax_glom(relab_ps, "ARG")

# Loop through the pairs
for (pair in label_case_pairs) {
  label_column <- pair$label
  case_value <- pair$case
  
  # Create a unique output prefix for each pair
  output_prefix <- paste0("siamcat_", label_column, "_", case_value)
  
  # Run the function
  siamcat_result <- create_siamcat_plot(psglom, label_column, case_value, output_prefix)
  
  # You can do additional processing with siamcat_result here if needed
  
  volcano.plot(siamcat_result)
    
  print(rownames(associations(siamcat_result))[associations(siamcat_result)$p.adj<0.05])
  # Print a message to indicate progress
  cat("Completed processing for", label_column, "with case", case_value, "\n")
}
```

## Industry

In this section industrial factor importance for changes in diversity are evaluated.

Show what are factors used.

```{r, print_fact}
colnames(sample_data(ps_amr_CSS))[7:12]
```
#### Industrial_wastewater_impact:

```{r, kruskal}
# remove NA values
ps_amr_ww_impact = subset_samples(ps_scaffolds_filtered, sample_data(ps_scaffolds_filtered)$Industrial_wastewater_impact != "None")


alpha_indexes_ww_impact <- estimate_richness(ps_amr_ww_impact, split = TRUE, c("Shannon", "Simpson", "InvSimpson"))
kruskal.test(alpha_indexes_ww_impact$Shannon ~ sample_data(ps_amr_ww_impact)$Industrial_wastewater_impact)
kruskal.test(alpha_indexes_ww_impact$Simpson ~ sample_data(ps_amr_ww_impact)$Industrial_wastewater_impact)
```


Find which pairs are important

```{r, dunes_amr}
dunn_results <- dunn.test(alpha_indexes_ww_impact$Shannon, 
          sample_data(ps_amr_ww_impact)$Industrial_wastewater_impact, 
          method="bh")
```


#### Industrial_wastewater_impact_from_food


```{r,kruskal_food}
# remove NA values
ps_amr_ww_impact = subset_samples(ps_scaffolds_filtered, sample_data(ps_scaffolds_filtered)$Industrial_wastewater_impact_from_food != "None")

alpha_indexes_ww_impact <- estimate_richness(ps_amr_ww_impact, split = TRUE, c("Shannon", "Simpson", "InvSimpson"))
kruskal.test(alpha_indexes_ww_impact$Shannon ~ sample_data(ps_amr_ww_impact)$Industrial_wastewater_impact_from_food)
kruskal.test(alpha_indexes_ww_impact$Simpson ~ sample_data(ps_amr_ww_impact)$Industrial_wastewater_impact_from_food)
```

Find which pairs are important

```{r, dunes_food}
dunn_results <- dunn.test(alpha_indexes_ww_impact$Shannon, 
          sample_data(ps_amr_ww_impact)$Industrial_wastewater_impact_from_food, 
          method="bh")
```


#### Binary Factor alpha diversity

Municipalities were classefied by Dairy_farming, Meat_production, Metal_processing and car washing facilites connected to wastewater system. 


```{r}
wilcox.test(alpha_indexes_amr$Shannon~sample_data(ps_scaffolds_filtered)$Dairy_farming, p.adjust.method = "BH")

wilcox.test(alpha_indexes_amr$Shannon~sample_data(ps_scaffolds_filtered)$Meat_production, p.adjust.method = "BH")

wilcox.test(alpha_indexes_amr$Shannon~sample_data(ps_scaffolds_filtered)$Metal_processing, p.adjust.method = "BH")

wilcox.test(alpha_indexes_amr$Shannon~sample_data(ps_scaffolds_filtered)$Washrooms, p.adjust.method = "BH")
```
```{r, wilcox_sipm}
wilcox.test(alpha_indexes_amr$Simpson~sample_data(ps_scaffolds_filtered)$Dairy_farming, p.adjust.method = "BH")

wilcox.test(alpha_indexes_amr$Simpson~sample_data(ps_scaffolds_filtered)$Meat_production, p.adjust.method = "BH")

wilcox.test(alpha_indexes_amr$Simpson~sample_data(ps_scaffolds_filtered)$Metal_processing, p.adjust.method = "BH")

wilcox.test(alpha_indexes_amr$Simpson~sample_data(ps_scaffolds_filtered)$Washrooms, p.adjust.method = "BH")
```

### Beta diversity industry


```{r, PERMANOVA}
# Microbial Community Diversity Analysis Tutorial with Phyloseq
# Permanova
# https://deneflab.github.io/MicrobeMiseq/demos/mothur_2_phyloseq.html

# remove NAs
ps_amr_ww_impact = subset_samples(ps_amr_CSS, sample_data(ps_amr_CSS)$Industrial_wastewater_impact != "None")
tax_bray_ww <- phyloseq::distance(ps_amr_ww_impact, method = "bray")

adonis2_rez<-adonis2(tax_bray_ww ~ sample_data(ps_amr_ww_impact)$Industrial_wastewater_impact)
print(adonis2_rez)
```



From Food

```{r, PERMANOVA2}
# remove NAs
ps_amr_ww_impact = subset_samples(ps_amr_CSS, sample_data(ps_amr_CSS)$Industrial_wastewater_impact_from_food != "None")

tax_bray_ww <- phyloseq::distance(ps_amr_ww_impact, method = "bray")

adonis2_rez<-adonis2(tax_bray_ww ~ sample_data(ps_amr_ww_impact)$Industrial_wastewater_impact_from_food)
print(adonis2_rez)


metadata_col = colnames(sample_data(ps_amr_CSS))[9:12]
```

```{r}
beta <- betadisper(tax_bray_ww, sample_data(ps_amr_ww_impact)$Industrial_wastewater_impact_from_food)
print(anova(beta))
plot(beta)
```


Show rest of the factors.

```{r, PERMANOVA3}
set.seed(7)

for (param in metadata_col){
# Using Bray-Curtis distance by default
if (param == "norm_factor" || param == "Sample" || param == "Date"){
    next
}

print(param)
formula_str <- paste("tax_bray_amr ~", param)  
adonis2_rez<-adonis2(as.formula(formula_str), data = data.frame(sample_data(ps_amr_CSS)))

# View results
print(adonis2_rez)

}

```

