---
title: "Wastewater microbiome in Latvia"
author: "Edgars Liepa"
date: "22/2/2025"
output:
  pdf_document: default
  html_document: default
---

Lets start with loading necessary libraries.

```{r, lib, warning=FALSE}
library("phyloseq")
library("vegan")
library("dplyr")
suppressPackageStartupMessages(library(microViz))
library("ggplot2")
```

Load metadata

```{r, paths}

METADATA_PATH =  "../sampleMetadata.csv"
metadata <- read.csv(METADATA_PATH, header = TRUE, colClasses = c(Date = "character", Dairy_farming="character", Meat_production="character", Metal_processing="character", Washrooms="character"))

```

## Bacterial abundance

Read summarized Kraken2 report biome file that was generated with [*kraken-biom*](https://github.com/smdabdoub/kraken-biom) tool from Kraken2 otputs

### reading sequencing rads

```{r, read_kraken_res}
merged_metagenomes <- import_biom("/home/edgars.liepa/NAS/bioinfo/edgars.liepa/Antibiotic_resistance/ww_publication/kraken_res/waste_water.biome")

# Clean Tax names
merged_metagenomes@tax_table@.Data <- substring(merged_metagenomes@tax_table@.Data, 4)
# Set collumn names
colnames(merged_metagenomes@tax_table@.Data)<- c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species")

# Fix Sample names
sample_names(merged_metagenomes) <- gsub("_k2_report$", "", sample_names(merged_metagenomes))

# Add metadata
sample_data(merged_metagenomes) <- sample_data(metadata)
 
```

Set Sample names to descriptive City names

```{r, rename_to_city_names}
# Create a named vector for mapping
name_mapping <- setNames(metadata$Sample, rownames(metadata))

# Use the mapping to rename the columns
sample_names(merged_metagenomes) <- name_mapping[sample_names(merged_metagenomes)]

```

### filtering

Subset bacteria only, since we are only interested in those.

```{r, filter, warning=FALSE}
merged_metagenomes <- subset_taxa(merged_metagenomes, Kingdom == "Bacteria")
```

Read counts in samples used in analysis

```{r, bar_mg, warning=FALSE}
plot_bar(merged_metagenomes)
```

Histogram of sample size distribution. We can see that one sample is cleare outlier

```{r, hist_mg}
hist(log10(sample_sums(merged_metagenomes)), breaks=50, main="Sample size distribution", xlab="Sample size (log10)", ylab="Frequency", col="#007c80")
```

```{r, save_hist_tax, include = FALSE}
# save plot

png("../plots/histogram_tax.png", width = 4.24, height = 4.86, units = "in", res = 300)

hist(log10(sample_sums(merged_metagenomes)), breaks=50, main="Sample size distribution", xlab="Sample size (log10)", ylab="Frequency", col="#007c80")

dev.off()

```

### Show Total sequence and OTU counts.

Total sequence count before filtering

```{r, counts_bact}
sum(sample_sums(merged_metagenomes))
```

Count of uniqe taxas reconstructed before filtering

```{r, counts_bact2}
dim(otu_table(merged_metagenomes))
```

#### Remove Outliers

Talsi 3, was compromised by an unexpected industrial event - a dairy company wastewater discharge that occurred during the sampling period. This contamination is clearly visible in the data as an abnormal spike in Lactobacillus helveticus abundance. Here it can be seen that low abundance of reads in Salaspils show increased relative diversity of most abundant Species.

Remove samples based on sample size distribution and outlier event on Salspils.2 and Talsi.3

```{r, rm_outlier}

ps_good = subset_samples(merged_metagenomes, sample_data(merged_metagenomes)$Sample != "Salaspils.2")
ps_good = subset_samples(ps_good, sample_data(ps_good)$Sample != "Talsi.3")

hist(log10(sample_sums(ps_good)), breaks=50, main="Sample size distribution", xlab="Sample size (log10)", ylab="Frequency", col="#007c80")

boxplot(sample_sums(ps_good), main="Sequencing depth across samples remove singletons", xlab="", ylab="Number of reads", col="#a6093d")
text(y=boxplot.stats(sample_sums(ps_good))$stats, labels=boxplot.stats(sample_sums(ps_good))$stats, x=1.25)
```

Difference between samples with largest and smallest sample deapth.

The max difference in sequencing depth is

```{r, filter_bact_after}
max(sample_sums(ps_good))/min(sample_sums(ps_good))
```

#### Filter low abundance tax

```{r, filter_low_abundance}
###
# Different options for filters
###

# Filter Singletons
ps_bact_filtered = filter_taxa(ps_good, function(x) sum(x) > 1, TRUE)
# Remove taxa not seen more than 3 times in at least 20% (9) of the samples. This protects against an OTU with small mean & trivially large C.V.
#ps_bact_filtered = filter_taxa(ps_bact, function(x) sum(x > 3) > (0.2*length(x)), TRUE)

# Filter OTUs to include only those with at least 7 reads in 5% (2) of the samples
#ps_bact_filtered = filter_taxa(merged_metagenomes, function(x) sum(x > 7) > (0.05 * length(x)), TRUE)

```

Difference between samples with largest and smallest sample deapth after filtering singletons.

```{r, filter_bact}
max(sample_sums(ps_bact_filtered))/min(sample_sums(ps_bact_filtered))
```

Phyloseq ARG scaffolds after filtering Print the dimensions of the filtered data

```{r, stats}
dim(data.frame(otu_table(ps_bact_filtered)))
```

Print the total sequence count after filtering Filtered sequence count

```{r, stats2}
sum(data.frame(otu_table(ps_bact_filtered)))
```

Calculate and print the percentage of sequences dropped from the original dataset

```{r, seq_drop}
original_seq_count = sum(data.frame(otu_table(ps_good)))
filtered_seq_count = sum(data.frame(otu_table(ps_bact_filtered)))
seq_dropped_percentage = ((original_seq_count - filtered_seq_count) / original_seq_count) * 100
```

Sequence % dropped from the dataset:

```{r, seq_drop2}
seq_dropped_percentage
```

```{r, box_after}

options(repr.plot.width=4, repr.plot.height=5)

boxplot(sample_sums(ps_bact_filtered), main="Sequencing depth across samples remove singletons", xlab="", ylab="Number of reads", col="#a6093d")
text(y=boxplot.stats(sample_sums(ps_bact_filtered))$stats, labels=boxplot.stats(sample_sums(ps_bact_filtered))$stats, x=1.25)
```

Sequence Count per Sample

```{r, seq_c}
sample_sums(ps_bact_filtered)
```

### Normalization

Normalization by subsampling (rarefaction)

```{r, rareCurv, warning=FALSE}
tab <- otu_table(ps_bact_filtered)
class(tab) <- "matrix"
tab <- t(tab)

rare <- rarecurve(tab, step=1000, lwd=2, ylab="OTU",  label=F)
# rare
```

rarefy samples if needed. Not used currently

```{r, rare}
# ps_rare = rarefy_even_depth(ps_bact_filtered, sample.size=5010685, replace=FALSE, rngseed=123, verbose=FALSE)
```

Normalization by cumulative sum scaling (CSS)

```{r, css, warning=FALSE}
ps_CSS =  microbiomeMarker::normalize(ps_bact_filtered, method="CSS")
```

### Tax Composition

```{r, tax_Genus, warning=FALSE}

# subset to genus
ps_bact_filtered_genus <- tax_glom(ps_bact_filtered, taxrank = "Genus")

ps_bact_filtered_genus <- ps_bact_filtered_genus %>% tax_fix(unknowns = c("Candidatus Nardonella"))

ps_bact_filtered_genus %>%
    ps_mutate(Group = factor(sample_data(ps_bact_filtered_genus)$City)) %>%
    tax_agg("Genus") %>%
    ps_seriate(dist = "bray", method = "OLO_ward") %>%
    comp_barplot(tax_level = "Genus", 
                 sample_order = sample_data(ps_bact_filtered_genus)[order(sample_data(ps_bact_filtered_genus)$Sample), ]$Sample, 
                 n_taxa = 10, 
                 label = "Sample", interactive = TRUE) +
    theme(legend.position = "bottom", axis.text.x = element_text(angle = 90, hjust = 1))+
    facet_grid(~Group, scales = "free", space = "free")

```

Show species composition plots with and without removed samples

```{r, tax_species, warning=FALSE}

# subset to species
ps_bact_filtered_species <- tax_glom(merged_metagenomes, taxrank = "Species")

## Fix species names
tax_table(ps_bact_filtered_species) <- tax_table(ps_bact_filtered_species) %>%
  as.data.frame() %>%
  mutate(Genus_species = case_when(
    is.na(Species) | Species == "" ~ Genus,
    TRUE ~ paste(Genus, Species, sep="_")
  )) %>%
  as.matrix()


species_all_samples <- ps_bact_filtered_species %>%
  ps_mutate(Group = factor(sample_data(ps_bact_filtered_species)$City)) %>%
  tax_agg("Genus_species") %>%
  ps_seriate(dist = "bray", method = "OLO_ward") %>%
  comp_barplot(
    tax_level = "Genus_species",
    sample_order = sample_data(ps_bact_filtered_species)[order(sample_data(ps_bact_filtered_species)$Sample), ]$Sample,
    n_taxa = 10,
    label = "Sample",
    interactive = FALSE
  ) +
  theme(
    legend.position = "bottom",
    axis.text.x =  element_blank(), # Increase x-axis text size
    axis.ticks.x =  element_blank(),
    axis.text.y = element_text(size = 12),
    axis.title = element_text(size = 14),
    legend.text = element_text(size = 12),
    strip.text = element_text(size = 10),
    guides(fill = guide_legend(nrow = 4))
  ) +
  facet_grid(~Group, scales = "free_x", space = "free_x")+ # scales and space free on x axis
  labs(y = "Relative Abundance", fill = "Species") # Change axis and legend titles

species_all_samples
```

```{r, save_tax_clean, echo=FALSE}
# Save as PNG
ggsave("../plots/species_all_samples_barplot.png", plot = species_all_samples, width = 12, height = 8, units = "in", dpi = 300)

# Save as SVG
ggsave("../plots/species_all_samples_barplot.svg", plot = species_all_samples, width = 12, height = 8, units = "in")
```


Diagram for publication.

```{r}
library(cowplot)
sample_sizes <- log10(sample_sums(merged_metagenomes))
  
  # Create the histogram using ggplot2
p1 <- ggplot(data.frame(size = sample_sizes), aes(x = size)) +
    geom_histogram(fill = "#007c80", color = "black", bins = 50) +
    labs(title = "Sample size distribution",
         x = "Sample size (log10)",
         y = "Frequency") +
    theme_classic() +
    theme(plot.title = element_text(size = 12, face = "bold"),
          plot.margin = unit(c(0.5, 0.5, 0.5, 0.5), "cm"))


species_all_samples


w_outliers <- plot_grid(
    p1, species_all_samples,
    labels = c("a", "b"),
    label_size = 16,
    ncol = 2,
    align = "v",
    rel_widths = c(0.75, 2)  # Adjust relative heights as needed
  )


```

```{r, save_tax_clean, echo=FALSE}
# Save as PNG
ggsave("../plots/w_outliers.jpg", plot = w_outliers, width = 18, height = 8, units = "in", dpi = 1200)

# Save as SVG
ggsave("../plots/w_outliers.svg", plot = w_outliers, width = 20, height = 8, units = "in")
```


Show aggregated species plot after samples are removed.

Top 20 most abundant genus distribution across municipalities.

```{r, tax_clean, warning=FALSE}
genus_plot <- ps_bact_filtered_genus %>% phyloseq::merge_samples(group = "City") %>%
    comp_barplot(tax_level = "Genus", 
                 n_taxa = 20)+
  theme(
    legend.position = "right",
    axis.text.x = element_text(angle = 90, hjust = 1, size = 12),
    axis.text.y = element_text(size = 12),
    axis.title = element_text(size = 12),
    legend.text = element_text(size = 12),
    guides(fill = guide_legend(ncol = 1))
  ) +
  labs(y = "Relative Abundance", fill = "Genus")

genus_plot

```

```{r, sve_genus plot, echo=FALSE}

# Save as PNG
ggsave("../plots/city_genus_barplot.jpg", plot = genus_plot, width = 12, height = 8, units = "in", dpi = 1200)

# Save as SVG
ggsave("../plots/city_genus_barplot.svg", plot = genus_plot, width = 12, height = 8, units = "in")

```

Top 15 species distribution across municipalities

```{r, species_plot, warning=FALSE}

ps_bact_filtered_species %>%
    phyloseq::merge_samples(group = "City") %>%
    comp_barplot(tax_level = "Genus_species", 
                 n_taxa = 15)+
    theme(axis.text.x = element_text(angle = 90, hjust = 1))+
  labs(y = "Relative Abundance", fill = "Species")
```

### Most abundant organisms

#### Genus

```{r, genera_count}

abu_table <- as.data.frame(otu_table(ps_bact_filtered_genus))
tax_table <- as.data.frame(tax_table(ps_bact_filtered_genus))
species_names <- tax_table$Genus
rownames(abu_table) <- tax_table$Genus


total_counts <- rowSums(abu_table)


species_abundance <- data.frame(
    Species = c(
        "Klebsiella", "Acinetobacter",
        "Pseudomonas", "Escherichia",
        "Citrobacter", "Aeromonas",
        "Enterobacter"
    )
)


species_abundance <- data.frame(
    Species = c(
        "Arcobacter", "Aeromonas", "Bacteroides", "Acinetobacter", "Lactobacillus", "Lactococcus", "Pseudoarcobacter", "Pseudomonas" , "Cloacibacterium", "Raoultella"
    )
)

# Calculate total counts and relative abundance
species_abundance$Total_Counts <- sapply(species_abundance$Species, 
    function(x) sum(total_counts[x]))

species_abundance

```


Calculate mean relative abundance and variance

```{r}

genus_transformed  = transform_sample_counts(ps_bact_filtered_genus, function(x) x / sum(x) )

abu_table <- as.data.frame(otu_table(genus_transformed))
tax_table <- as.data.frame(tax_table(genus_transformed))
species_names <- tax_table$Genus
rownames(abu_table) <- tax_table$Genus

variance <- apply(abu_table, 1, var)
mean_percentage <- apply(abu_table, 1, mean) * 100
std_dev <- sqrt(apply(abu_table, 1, var)) * 100



# Select genera of interest
genus_abundance <- data.frame(
  Genus = c("Arcobacter", "Aeromonas", "Bacteroides", "Acinetobacter", "Citrobacter","Escherichia", "Enterobacter", 
                     "Lactobacillus", "Lactococcus", "Pseudoarcobacter", 
                     "Pseudomonas", "Cloacibacterium", "Raoultella", "Klebsiella")
)

genus_abundance$mean_percentage <- sapply(genus_abundance$Genus, function(genus)
          unname(mean_percentage[genus]))

genus_abundance$std_dev <- sapply(genus_abundance$Genus, function(genus)
          unname(std_dev[genus]))


# Round relative abundance to 2 decimal places
genus_abundance$mean_percentage <- round(genus_abundance$mean_percentage, 4)
genus_abundance$std_dev <- round(genus_abundance$std_dev, 4)


genus_abundance


# Format output with both mean percentage and standard deviation
formatted_output <- sapply(genus_abundance["Genus"], function(genus) {
  sprintf("%s (%.3f %% +/- %.3f %%)", 
          genus, 
          mean_percentage[genus], 
          std_dev[genus])
})

# Display formatted output
formatted_output


```

Now we look at bacteria that look like dominating in some cities or are outliers.  
 
```{r, genus_outliers, warning=FALSE}
sample_df <- data.frame(
  value = as.numeric((abu_table["Streptococcus",]))*100,
  city = colnames(abu_table)
)

ggplot(sample_df, aes(x=as.factor(city), y=value))+geom_bar(stat = "identity")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))+
  geom_hline(yintercept = mean_percentage["Streptococcus"], color = "red", linewidth = 1) +
  geom_text(aes(x = 1, y = mean_percentage["Streptococcus"]), 
          label = paste("Mean:", round(mean_percentage["Streptococcus"], 4)),
          color = "red", hjust = 0, vjust = -1.5) +
  labs(y = "Relative Abundance", x = "City", title =  "abundance ")

print(abu_table["Streptococcus",]*100)
```


```{r, genus_outliers_madona, warning=FALSE}
std_dev["Streptococcus"]
mean_percentage["Streptococcus"]

abu_table["Streptococcus","Madona.2"]*100
```

Valmiera had the highest levels of Cloacibacterium (Valmiera mean 13.71 % +/- 5.54 %), comprising more than a quarter of the genera. This was consistent across all sample repeats and indicated a clear difference from other cities.


```{r, genus_outliers2, warning=FALSE}
sample_df <- data.frame(
  value = as.numeric((abu_table["Cloacibacterium",]))*100,
  city = colnames(abu_table)
)

ggplot(sample_df, aes(x=as.factor(city), y=value))+geom_bar(stat = "identity")+
  geom_hline(yintercept = mean_percentage["Cloacibacterium"], color = "red", linewidth = 1) +
 geom_text(aes(x = 1, y = mean_percentage["Cloacibacterium"]), 
          label = paste("Mean:", round(mean_percentage["Cloacibacterium"], 4)),
          color = "red", hjust = 0, vjust = -1) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))+
  labs(y = "Relative Abundance", x = "City", title =  "Cloacibacterium abundance")


print(abu_table["Cloacibacterium",]*100)
```

```{r, genus_outliers3, warning=FALSE}
# Check outlier cities

print(
  paste(
    "Cloacibacterium mean realative abundance:",
    round(mean_percentage["Cloacibacterium"],2), "% +/-",
    round(std_dev["Cloacibacterium"],2),"%"
  )
)

print(
  paste(
    "Madona.3 relative abundance ",
    round(abu_table["Cloacibacterium","Madona.3"]*100,2),
    "%"
  )
)

print(
  paste(
    "Valmiera mean realative abundance:",
    round(apply(abu_table["Cloacibacterium",c("Valmiera.1","Valmiera.2","Valmiera.3")], 1, mean) * 100,2),
    "% +/-",
    round(sqrt(apply(abu_table["Cloacibacterium",c("Valmiera.1","Valmiera.2","Valmiera.3")], 1, var)) * 100,2),
    "%"
  )
)

```


#### Species 

Show most combined read count for top 15 bacteria.

```{r, top_tax}
abu_table <- as.data.frame(otu_table(ps_bact_filtered_species))
tax_table <- as.data.frame(tax_table(ps_bact_filtered_species))
rownames(abu_table) <- tax_table$Genus_species

top_organisms <- sort(rowSums(abu_table), decreasing = TRUE) %>% head(n = 15)

top_organisms
```


Calculate the mean relative abundance and standard deviation of the top 15 species

```{r, top_tax2}

genus_transformed  = transform_sample_counts(ps_bact_filtered_species, function(x) x / sum(x) )
abu_table <- as.data.frame(otu_table(genus_transformed))
tax_table <- as.data.frame(tax_table(genus_transformed))
species_names <- tax_table$Genus_species
rownames(abu_table) <- tax_table$Genus_species

mean_percentage <- apply(abu_table, 1, mean) * 100
std_dev <- sqrt(apply(abu_table, 1, var)) * 100

top_species <- sort(mean_percentage, decreasing = TRUE) %>% head(n = 15) %>% as.data.frame %>% rownames

# Format output with both mean percentage and standard deviation
formatted_output <- sapply(top_species, function(species) {
  sprintf("%s (%.2f%% +/- %.2f%%)", 
          species, 
          unname(mean_percentage[species]), 
          unname(std_dev[species])
          )
})

# Display formatted output
formatted_output


```



#### Bacteria with clinical relevance

Calculate the mean relative abundance and standard deviation for ESKAPE pathogens: 

The ESKAPE pathogens — Enterococcus faecium, Staphylococcus aureus, Klebsiella pneumoniae, Acinetobacter baumannii, 
Pseudomonas aeruginosa and Enterobacter spp. — are identified as critical multidrug-resistant bacteria for which effective therapies are needed. 


```{r, bact_dang}

species_abundance <- data.frame(
    Species = c(
         "Klebsiella_pneumoniae", "Klebsiella_huaxiensis",
        "Acinetobacter_baumannii", "Acinetobacter_johnsonii",
        "Pseudomonas_aeruginosa", "Pseudomonas_alcaligenes",
        "Escherichia_coli", "Citrobacter_freundii",
        "Citrobacter_portucalensis", "Citrobacter_braakii",
        "Aeromonas_caviae", "Aeromonas_dhakensis",
        "Aeromonas_veronii", "Enterobacter_cloacae",
        "Enterococcus_faecium", "Staphylococcus_aureus"
    )
)

# Calculate total counts and relative abundance
species_abundance$Total_Counts <- sapply(species_abundance$Species, 
    function(x) sum(total_counts[x]))

species_abundance$mean_percentage <- sapply(species_abundance$Species, function(species)
          unname(mean_percentage[species]))

species_abundance$std_dev <- sapply(species_abundance$Species, function(species)
          unname(std_dev[species]))


# Round relative abundance to 4 decimal places
species_abundance$mean_percentage <- round(species_abundance$mean_percentage, 4)
species_abundance$std_dev <- round(species_abundance$std_dev, 4)


species_abundance

sapply(species_abundance["Species"], function(species) {
  sprintf("%s (%.3f %% +/- %.3f %%)", 
          species, 
          unname(mean_percentage[species]), 
          unname(std_dev[species])
          )
})
```


```{r, species_outliers2, warning=FALSE}
sample_df <- data.frame(
  value = as.numeric((abu_table["Arcobacter_cryaerophilus",]))*100,
  city = colnames(abu_table)
)

ggplot(sample_df, aes(x=as.factor(city), y=value))+geom_bar(stat = "identity")+
  geom_hline(yintercept = mean_percentage["Arcobacter_cryaerophilus"], color = "red", linewidth = 1) +
 geom_text(aes(x = 1, y = mean_percentage["Arcobacter_cryaerophilus"]), 
          label = paste("Mean:", round(mean_percentage["Arcobacter_cryaerophilus"], 4)),
          color = "red", hjust = 0, vjust = -1) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))+
  labs(y = "Relative Abundance", x = "City", title =  "Arcobacter_cryaerophilus abundance")


print(abu_table["Arcobacter_cryaerophilus",]*100)
```

```{r, species_outliers3, warning=FALSE}
sample_df <- data.frame(
  value = as.numeric((abu_table["Cloacibacterium_normanense",]))*100,
  city = colnames(abu_table)
)

ggplot(sample_df, aes(x=as.factor(city), y=value))+geom_bar(stat = "identity")+
  geom_hline(yintercept = mean_percentage["Cloacibacterium_normanense"], color = "red", linewidth = 1) +
 geom_text(aes(x = 1, y = mean_percentage["Cloacibacterium_normanense"]), 
          label = paste("Mean:", round(mean_percentage["Cloacibacterium_normanense"], 4)),
          color = "red", hjust = 0, vjust = -1) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))+
  labs(y = "Relative Abundance", x = "City", title =  "Cloacibacterium_normanense abundance")


print(abu_table["Cloacibacterium_normanense",]*100)
```
```{r}
mean_percentage["Cloacibacterium_normanense"]
std_dev["Cloacibacterium_normanense"]

print(
  paste(
    "Madona.3 relative abundance ",
    abu_table["Cloacibacterium_normanense","Madona.3"]*100
  )
)

print(
  paste(
    "Valmiera mean realative abundance:",
    apply(abu_table["Cloacibacterium_normanense",c("Valmiera.1","Valmiera.2","Valmiera.3")], 1, mean) * 100,
    " %"
  )
)

print(
  paste(
    "Valmiera standart deviation:",
    sqrt(apply(abu_table["Cloacibacterium_normanense",c("Valmiera.1","Valmiera.2","Valmiera.3")], 1, var)) * 100,
    " %"
  )
)
```


```{r, lacto_plot, warning=FALSE}

sample_df <- data.frame(
  value = as.numeric((abu_table["Lactococcus_raffinolactis",]))*100,
  city = colnames(abu_table)
)

ggplot(sample_df, aes(x=as.factor(city), y=value))+geom_bar(stat = "identity")+
  geom_hline(yintercept = mean_percentage["Lactococcus_raffinolactis"], color = "red", linewidth = 1) +
 geom_text(aes(x = 1, y = mean_percentage["Lactococcus_raffinolactis"]), 
          label = paste("Mean:", round(mean_percentage["Lactococcus_raffinolactis"], 4)),
          color = "red", hjust = 0, vjust = -1) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))+
  labs(y = "Relative Abundance", x = "City", title =  "Lactococcus_raffinolactis abundance")


print(abu_table["Lactococcus_raffinolactis",]*100)
```


```{r}
mean_percentage["Lactococcus_raffinolactis"]
std_dev["Cloacibacterium_normanense"]

print(
  paste(
    "Madona.3 relative abundance ",
    abu_table["Lactococcus_raffinolactis","Madona.3"]*100
  )
)


######### Smiltene #########

print(
  paste(
    "Smiltene mean realative abundance:",
    round(apply(abu_table["Lactococcus_raffinolactis",c("Smiltene.1","Smiltene.2","Smiltene.3")], 1, mean) * 100,2),
    "% +/-", 
    round(
        sqrt(apply(abu_table["Lactococcus_raffinolactis",c("Smiltene.1","Smiltene.2","Smiltene.3")], 1, var)) * 100,
      2),
    "%"
  )
)

######### Tukums #########

print(
  paste(
    "Tukums mean realative abundance:",
    round(apply(abu_table["Lactococcus_raffinolactis",c("Tukums.1","Tukums.2","Tukums.3")], 1, mean) * 100,2),
    "% +/-",
    round(sqrt(apply(abu_table["Lactococcus_raffinolactis",c("Tukums.1","Tukums.2","Tukums.3")], 1, var)) * 100,2),
    "%"
  )
)

######### Talsi #########

print(
  paste(
    "Lactococcus_raffinolactis Talsi mean realative abundance:",
    round(apply(abu_table["Lactococcus_raffinolactis",c("Talsi.1","Talsi.2","Talsi.3")], 1, mean) * 100,2),
    "% +/-",
    round(sqrt(apply(abu_table["Lactococcus_raffinolactis",c("Talsi.1","Talsi.2","Talsi.3")], 1, var)) * 100,2),
    "%"
  )
)
```


## City

### Alpha Diversity

Calculations of alpha diversity Shannon and InverSimpson differences between municipality samples.

```{r, alpha_div_pl}
#alpha diversity boxplot graph

alpha_plot <- plot_richness(ps_bact_filtered, x = "City",
              measures=c("Shannon", "InvSimpson"), nrow = 2, sortby = "Shannon") + 
              geom_boxplot() + geom_point(size=1, alpha=0.5)+
  theme(
    legend.position = "right",
    axis.text.x = element_text(angle = 90, hjust = 1, size = 14),
    axis.text.y = element_text(size = 12),
    axis.title = element_text(size = 14),
    legend.text = element_text(size = 16),
    strip.text = element_text(size = 10),
    guides(fill = guide_legend(ncol = 1))
  )

alpha_plot
```

```{r, save_alpha_city_plot, echo=FALSE}
# Save as PNG
ggsave("../plots/richness_bact.png", plot = alpha_plot, width = 6, height = 6, units = "in", dpi = 300)

# Save as SVG
ggsave("../plots/richness_bact.svg", plot = alpha_plot, width = 6, height = 6, units = "in")

```

Use Kruskal Wallis test to check if differences between municipality Shannon and InvSimpson indexes are significant.

```{r, kruskal_city}
alpha_indexes_bact <- estimate_richness(ps_bact_filtered, split = TRUE, c("Shannon", "Simpson", "InvSimpson"))

kruskal.test(alpha_indexes_bact$Shannon ~ sample_data(ps_bact_filtered)$City)
kruskal.test(alpha_indexes_bact$InvSimpson ~ sample_data(ps_bact_filtered)$City)

```

```{r}
min(alpha_indexes_bact$InvSimpson)
max(alpha_indexes_bact$InvSimpson)
min(alpha_indexes_bact$Shannon)
max(alpha_indexes_bact$Shannon)
```

Calculate Dunes Test to pinpoint which specific means between municipalities are most significant.

```{r}
library(dunn.test)
dunn_results <- dunn.test(alpha_indexes_bact$InvSimpson, 
          sample_data(ps_bact_filtered)$City, 
          method="bh")
```

```{r}
dunn_results$P.adjusted[dunn_results$P.adjusted < 0.05]
dunn_results$chi2
dunn_results$Z[dunn_results$P.adjusted < 0.05]
dunn_results$comparisons[dunn_results$P.adjusted < 0.05]
```

```{r}
dunn_results <- dunn.test(alpha_indexes_bact$Shannon, 
          sample_data(ps_bact_filtered)$City, 
          method="bh")
```

```{r}
dunn_results$P.adjusted[dunn_results$P.adjusted < 0.05]
dunn_results$chi2
dunn_results$Z[dunn_results$P.adjusted < 0.05]
dunn_results$comparisons[dunn_results$P.adjusted < 0.05]
```


### Beta Diversity

Now we create NMDS Bray–Curtis distance ordination plot, to discover how similar are cities between each other.

```{r, beta_div_ord}
theme_set(theme_bw())

#https://joey711.github.io/phyloseq/plot_ordination-examples.html

GP.ord <- ordinate(ps_CSS, "NMDS", "bray")
GP.ord
```

```{r, nmds_plot}

p1 = plot_ordination(ps_CSS, GP.ord, type="Sample", color="City") + geom_point(size=6)+
  geom_polygon(aes(fill=City), alpha = 1/2)+
  theme(
    legend.position = "right",
    axis.text.x = element_text(angle = 90, hjust = 1, size = 14),
    axis.text.y = element_text(size = 14),
    axis.title = element_text(size = 14),
    legend.text = element_text(size = 19),
    legend.title = element_text(size = 18)
  ) 

p1

```

```{r, save_nmds_plot, city, echo=FALSE}
# Save as PNG
ggsave("../plots/nmds_beta.png", plot = p1, width = 12, height = 8, units = "in", dpi = 300)

# Save as SVG
ggsave("../plots/nmds_beta.svg", plot = p1, width = 12, height = 8, units = "in")
```


```{r}

alpha_beta_bact <- plot_grid(
    alpha_plot, p1,
    labels = c("a", "b"),
    label_size = 16,
     rel_widths = c(1, 1.5)
  )
```

```{r, save_nmds_plot, city, echo=FALSE}
# Save as PNG
ggsave("../plots/Figure_4.jpg", plot = alpha_beta_bact, width = 16, height = 8, units = "in", dpi = 1200)

# Save as SVG
ggsave("../plots/Figure_4.tiff", plot = alpha_beta_bact, width = 16, height = 8, units = "in", dpi = 1200)
```

Now calculate Bray–Curtis distance again to use for Analysis of similarities (ANOSIM) and permutational multivariate analysis of variance using distance matrices (PERMANOVA).

```{r, tax_b}
tax_bray <- phyloseq::distance(ps_CSS, method = "bray")
```

```{r, anosim_city}

anosim(tax_bray, sample_data(ps_CSS)$City, distance = "bray", permutations = 999)

```

```{r, adonis_city}

adonis2_rez<-adonis2(tax_bray ~ sample_data(ps_CSS)$City)
print(adonis2_rez)
```

Lets calculate beta dispersion to verify similarity analysis results between municipalities.

```{r, beta_city_plot}
beta <- betadisper(tax_bray, sample_data(ps_CSS)$City)
print(anova(beta))
plot(beta)

```

## Hospital

One of the main research questions were how does the hospital influence local waste-water microbiome and resistance gene pool?

First we compare differences between sample alpha diversity measures in samples with and without regional Hospital facilities.

```{r, kruskal_regional}
wilcox.test(alpha_indexes_bact$Shannon~sample_data(ps_bact_filtered)$Regional_Hospital,
                     p.adjust.method = "BH")

wilcox.test(alpha_indexes_bact$InvSimpson~sample_data(ps_bact_filtered)$Regional_Hospital,
                     p.adjust.method = "BH")

```

Here differences in municipalities with regional hospital between Shannon and InvSimpson indices were not found.

#### Hospital Types

```{r, kruskal_hospital}
kruskal.test(alpha_indexes_bact$Shannon ~ sample_data(ps_bact_filtered)$Hospital_Type)
```

Display Shannon and InvSimpson values in Plot 

```{r, alpha_hospital}

plot_richness(ps_bact_filtered_species, x = "Hospital_Type",
              measures=c("Shannon", "InvSimpson"), nrow = 2, sortby = "Shannon") + 
              geom_boxplot() + geom_point(size=1, alpha=0.5)   + labs(
    title = "Species Hospital type diversity")
```

Now NMDS bray-curtis beta diversity plot is create where samples with and without regional hospitals are circled with ellipse. Since not all health-care institutions are regional hospital, different levels of hospital types are colored in a plot to look weather these institutions cluster together based on their type.

```{r, beta_hospital}
GP.ord2 <- ordinate(ps_CSS, "NMDS", "bray")
p1 = plot_ordination(ps_CSS, GP.ord2, type="samples", color = "Hospital_Type", title="Hospital_Type bray-distance NMDS")+
 stat_ellipse(aes(linetype = Regional_Hospital, colour = Regional_Hospital)) + geom_point(size=3)

print(p1)



```

```{r adonis_hs_type}
adonis2_rez<-adonis2(tax_bray ~ sample_data(ps_CSS)$Hospital_Type)
print(adonis2_rez)

```

Show BETADISPERSION for Hospital_Type

```{r, beta_hs_type}
beta <- betadisper(tax_bray, sample_data(ps_CSS)$Hospital_Type)
print(anova(beta))
plot(beta)
```

And now look which types are significantly different.

```{r, dunes_bact_hospital}

dunn_results <- dunn.test(alpha_indexes_bact$Shannon, 
          sample_data(ps_bact_filtered)$Hospital_Type, 
          method="bh")

dunn_results$P.adjusted[dunn_results$P.adjusted < 0.05]
dunn_results$chi2
dunn_results$Z[dunn_results$P.adjusted < 0.05]
dunn_results$comparisons[dunn_results$P.adjusted < 0.05]
```

## Industry

Lastly municipalities have different industrious that are variable between cities and are connected to wastewater system.

Municipalities were categorized by Industrial_wastewater_impact and different idustrise that are in the municipality.

Industrial_wastewater_impact significance:

```{r, Kruskal_industry}

ps_tax_ww_impact = subset_samples(ps_bact_filtered, sample_data(ps_bact_filtered)$Industrial_wastewater_impact != "None")

alpha_indexes_ww_impact <- estimate_richness(ps_tax_ww_impact, split = TRUE, c("Shannon", "Simpson", "InvSimpson"))

kruskal.test(alpha_indexes_ww_impact$Shannon ~ sample_data(ps_tax_ww_impact)$Industrial_wastewater_impact)


dunn_results <- dunn.test(alpha_indexes_ww_impact$Shannon, 
          sample_data(ps_tax_ww_impact)$Industrial_wastewater_impact, 
          method="bh")

```

Industrial_wastewater_impact_from_food:

```{r, k_industry_2}

kruskal.test(alpha_indexes_ww_impact$Shannon ~ sample_data(ps_tax_ww_impact)$Industrial_wastewater_impact_from_food)

kruskal.test(alpha_indexes_ww_impact$InvSimpson ~ sample_data(ps_tax_ww_impact)$Industrial_wastewater_impact_from_food)


dunn_results <- dunn.test(alpha_indexes_ww_impact$Shannon, 
          sample_data(ps_tax_ww_impact)$Industrial_wastewater_impact_from_food, 
          method="bh")


```

```{r, wilcox_industry_bact}
wilcox.test(alpha_indexes_ww_impact$Shannon~sample_data(ps_tax_ww_impact)$Dairy_farming, p.adjust.method = "BH")

wilcox.test(alpha_indexes_ww_impact$Shannon~sample_data(ps_tax_ww_impact)$Meat_production, p.adjust.method = "BH")

wilcox.test(alpha_indexes_ww_impact$Shannon~sample_data(ps_tax_ww_impact)$Metal_processing, p.adjust.method = "BH")

wilcox.test(alpha_indexes_ww_impact$Shannon~sample_data(ps_tax_ww_impact)$Washrooms, p.adjust.method = "BH")
```

```{r, wilxon_simp}

wilcox.test(alpha_indexes_ww_impact$Simpson~sample_data(ps_tax_ww_impact)$Dairy_farming, p.adjust.method = "BH")

wilcox.test(alpha_indexes_ww_impact$Simpson~sample_data(ps_tax_ww_impact)$Meat_production, p.adjust.method = "BH")

wilcox.test(alpha_indexes_ww_impact$Simpson~sample_data(ps_tax_ww_impact)$Metal_processing, p.adjust.method = "BH")

wilcox.test(alpha_indexes_ww_impact$Simpson~sample_data(ps_tax_ww_impact)$Washrooms, p.adjust.method = "BH")
```


### Beta diversity

Industrial_wastewater_impact

```{r, PERMANOVA_industry_tax}

ps_tax_ww_impact = subset_samples(ps_CSS, sample_data(ps_CSS)$Industrial_wastewater_impact != "None")

tax_bray_ww <- phyloseq::distance(ps_tax_ww_impact, method = "bray")

adonis2_rez<-adonis2(tax_bray_ww ~ sample_data(ps_tax_ww_impact)$Industrial_wastewater_impact)
print(adonis2_rez)

```

From Food

```{r, food_industr}

ps_tax_ww_impact = subset_samples(ps_CSS, sample_data(ps_CSS)$Industrial_wastewater_impact_from_food != "None")

tax_bray_ww <- phyloseq::distance(ps_tax_ww_impact, method = "bray")

adonis2_rez<-adonis2(tax_bray_ww ~ sample_data(ps_tax_ww_impact)$Industrial_wastewater_impact_from_food)
print(adonis2_rez)
```

PERMANOVA test for industrial variables.

```{r, loop_perm}
set.seed(7)

# take rest of the variables that we are looking at
metadata_col = colnames(sample_data(ps_CSS))[9:12]


for (param in metadata_col){
# Using Bray-Curtis distance by default
if (param == "norm_factor" || param == "Sample" || param == "Date"){
    next
}

print(param)
formula_str <- paste("tax_bray ~", param)  
adonis2_rez<-adonis2(as.formula(formula_str), data = data.frame(sample_data(ps_CSS)))

# View results
print(adonis2_rez)

}

```
